stack_test.go

```diff
@@ -0,0 +1,198 @@
+package gapi
+
+import (
+	"testing"
+)
+
+const (
+	getStackJSON    = `{"id": 1,"slug": "mystack"}`
+	createStackJSON = `{"id": 1,"slug": "mystack"}`
+	getStacksJSON   = `
+{
+	"items": [
+		{
+			"id": 1,
+			"orgId": 1,
+			"orgSlug": "myorg",
+			"orgName": "MyOrg",
+			"name": "mystack.grafana.net",
+			"url": "https://mystack.grafana.net",
+			"slug": "mystack",
+			"version": "stable",
+			"description": "My amazing stack",
+			"status": "active",
+			"gateway": "istio",
+			"createdAt": "2021-12-22T14:02:46.000Z",
+			"createdBy": "xyz",
+			"updatedAt": null,
+			"updatedBy": "",
+			"trial": 0,
+			"trialExpiresAt": null,
+			"clusterId": 666,
+			"clusterSlug": "prod-eu-west-0",
+			"clusterName": "prod-eu-west-0",
+			"plan": "gcloud",
+			"planName": "Grafana Cloud",
+			"billingStartDate": "2021-12-22T14:02:46.000Z",
+			"billingEndDate": null,
+			"billingActiveUsers": 0,
+			"currentActiveUsers": 1,
+			"currentActiveAdminUsers": 1,
+			"currentActiveEditorUsers": 0,
+			"currentActiveViewerUsers": 0,
+			"dailyUserCnt": 0,
+			"dailyAdminCnt": 0,
+			"dailyEditorCnt": 0,
+			"dailyViewerCnt": 0,
+			"billableUserCnt": 1,
+			"dashboardCnt": 6,
+			"datasourceCnts": {},
+			"userQuota": 10,
+			"dashboardQuota": -1,
+			"alertQuota": -1,
+			"ssl": true,
+			"customAuth": true,
+			"customDomain": true,
+			"support": true,
+			"runningVersion": "8.3.3 (commit: 30bb7a93c, branch: HEAD)",
+			"machineLearning": 0,
+			"hmInstancePromId": 112233,
+			"hmInstancePromUrl": "https://prometheus-prod-01-eu-west-0.grafana.net",
+			"hmInstancePromName": "mystack-prom",
+			"hmInstancePromStatus": "active",
+			"hmInstancePromCurrentUsage": 11111,
+			"hmInstancePromCurrentActiveSeries": 222222,
+			"hmInstanceGraphiteId": 111111,
+			"hmInstanceGraphiteUrl": "https://graphite-prod-01-eu-west-0.grafana.net",
+			"hmInstanceGraphiteName": "mystack-graphite",
+			"hmInstanceGraphiteType": "graphite-v5",
+			"hmInstanceGraphiteStatus": "active",
+			"hmInstanceGraphiteCurrentUsage": 0,
+			"hlInstanceId": 121416,
+			"hlInstanceUrl": "https://logs-prod-eu-west-0.grafana.net",
+			"hlInstanceName": "mystack-logs",
+			"hlInstanceStatus": "active",
+			"hlInstanceCurrentUsage": 0,
+			"amInstanceId": 654321,
+			"amInstanceName": "mystack-alerts",
+			"amInstanceUrl": "https://alertmanager-eu-west-0.grafana.net",
+			"amInstanceStatus": "active",
+			"amInstanceGeneratorUrl": "https://mystack.grafana.net",
+			"htInstanceId": 123456,
+			"htInstanceUrl": "https://tempo-eu-west-0.grafana.net",
+			"htInstanceName": "mystack-traces",
+			"htInstanceStatus": "active",
+			"regionId": 3,
+			"regionSlug": "eu"
+			}
+	]
+}
+`
+)
+
+func TestStacks(t *testing.T) {
+	server, client := gapiTestTools(t, 200, getStacksJSON)
+	defer server.Close()
+
+	stacks, err := client.Stacks()
+
+	if err != nil {
+		t.Fatalf("expected error to be nil; got: %s", err.Error())
+	}
+
+	actualItemCound := len(stacks.Items)
+	expectedItemCount := 1
```

Comment:
`len(getStacksJSON.Items)` perhaps?

Reply:
Its a string object so it can't be used as you suggested. I agree with your point though. 
I'll see if I can serialise the string into an object.

Reply:
serialize: object to string.
deserialize: string to object :D
`getStacksJSON.Items` is a string? 🤔

Reply:
ahh, damn, just saw backticks.

Reply:
yes, i think we sound not compare strings. keep test data as object, deserialize response and compare properties, fields etc. at least that is what i would do 🤷 :D

Reply:
Good catch, I have updated the code.